common/types/utils.go

```diff
@@ -63,3 +64,12 @@ func Intersection[T comparable](arrays ...[]T) []T {
 
 	return intersection
 }
+
+// SafePow implements a deterministic power function
+// Go's math.Pow() doesn't guarantee determinism when executed on different hardwares
+func SafePow(base uint64, exponent uint64) uint64 {
+	baseInt := sdk.NewIntFromUint64(base)
+	baseDec := sdk.NewDecFromInt(baseInt)
```

Comment:
why not use NewDecWithPrec? we have examples how to creade Dec from uint without creating an Int

Reply:
done. see https://github.com/lavanet/lava/commit/157c0df6080ba5ebb599d47c6873aca506b82b24

Reply:
corrected it. see https://github.com/lavanet/lava/commit/9fdaee7f74bb1c371aebabd65334ee396226e6f9